crates/router_env/src/request_id.rs

```diff
@@ -0,0 +1,365 @@
+    /// the same value as `to_string()` and `as_str()`.
+    pub fn as_hyphenated(&self) -> &str {
+        &self.0
+    }
+}
+
+impl RequestIdentifier {
+    /// Create a request ID middleware with a custom header name.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use router_env::RequestIdentifier;
+    ///
+    /// // Use any custom header name for request ID extraction
+    /// let middleware = RequestIdentifier::new("x-request-id");
+    /// ```
+    #[must_use]
+    pub fn new(header_name: &str) -> Self {
+        Self {
+            header_name: header_name.to_string(),
+            use_incoming_id: IdReuse::default(),
+        }
+    }
+
+    /// Configure how incoming request ID headers should be handled.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use router_env::{RequestIdentifier, IdReuse};
+    ///
+    /// let middleware = RequestIdentifier::new("x-request-id")
+    ///     .use_incoming_id(IdReuse::IgnoreIncoming);
+    /// ```
+    #[must_use]
+    pub fn use_incoming_id(self, use_incoming_id: IdReuse) -> Self {
+        Self {
+            use_incoming_id,
+            ..self
+        }
+    }
+}
+
+impl<S, B> Transform<S, ServiceRequest> for RequestIdentifier
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = ActixError>,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = S::Response;
+    type Error = S::Error;
+    type Transform = RequestIdMiddleware<S>;
+    type InitError = ();
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        let header_name = HeaderName::from_str(&self.header_name).unwrap_or_else(|_| {
+            tracing::error!(
+                "Invalid header name '{}', using fallback 'x-request-id'",
+                self.header_name
+            );
+            HeaderName::from_static("x-request-id")
+        });
+
+        ready(Ok(RequestIdMiddleware {
+            service,
+            header_name,
+            use_incoming_id: self.use_incoming_id,
+        }))
+    }
+}
+
+#[allow(clippy::type_complexity)]
+impl<S, B> Service<ServiceRequest> for RequestIdMiddleware<S>
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = ActixError>,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = S::Response;
+    type Error = S::Error;
+    type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>>>>;
+
+    fn poll_ready(&self, ctx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        self.service.poll_ready(ctx)
+    }
+
+    fn call(&self, request: ServiceRequest) -> Self::Future {
+        let header_name = self.header_name.clone();
+
+        // Capture incoming request ID for logging
+        let incoming_request_id = request.headers().get(&header_name).cloned();
+
+        // Extract from header or generate UUID v7 - simple and direct
```

Comment:
Can we improve such AI comments please, I don't think "simple and direct" adds any value here.